tools/PI/DevHome.PI/ViewModels/ExpandedViewControlViewModel.cs

```diff
@@ -236,23 +236,19 @@ public void NavigateTo(Type viewModelType)
 
     public void Navigate()
     {
-        var navigationService = Application.Current.GetService<INavigationService>();
-        navigationService.NavigateTo(Links[SelectedNavLinkIndex]?.PageViewModel?.FullName!);
+        if (SelectedNavLinkIndex != -1)
+        {
+            var navigationService = Application.Current.GetService<INavigationService>();
+            navigationService.NavigateTo(Links[SelectedNavLinkIndex]?.PageViewModel?.FullName!);
+        }
     }
 
     public void NavigateToSettings(string viewModelType)
     {
         // Because the Settings item isn't part of our NavLink list, when the user selects Settings,
         // we need to move the list selection so that when they subsequently select an item from
         // the NavLinks, we'll navigate to the correct page even if that was the previously-selected item.
-        if (SelectedNavLinkIndex == 0)
-        {
-            SelectedNavLinkIndex = Links.Count - 1;
-        }
-        else
-        {
-            SelectedNavLinkIndex = 0;
-        }
```

Comment:
Hmm... I tried this before, but it throws in some scenarios.

Reply:
Do you remember what scenarios? After adding the -1 check in Navigate() above, I haven't seen any issues.

The code as-is is broken in the case where we're not attached to any process, because Links.Count=1... so we're always setting SelectedNavLinkIndex to 0.

Not only does it jam the UI (for the reasons described above) but setting this to 0 (even if SelectedNavLinkIndex is already 0) ends up triggering another Navigate call, and so after navigating to the settings page we immediately navigate to page 0, navigating away from the settings page.